src/shared/errors/errors.go

```diff
@@ -10,6 +10,10 @@ func New(text string) error {
 	return bugsnagerrors.New(text, 1)
 }
 
+func NewCustomError(text string) error {
```

Comment:
```suggestion
func NewSentinelError(text string) error {
```
Align terminology with common terminology:
https://go.dev/blog/go1.13-errors#:~:text=Sometimes%20we%20compare%20an%20error%20to%20a%20known%20sentinel%20value